packages/core/core-flows/src/definition/cart/workflows/complete-cart.ts

```diff
@@ -32,17 +22,25 @@ export const completeCartWorkflow = createWorkflow(
       list: false,
     })
 
+    const paymentSession = validateCartPaymentsStep({ cart })
+
+    // Question: While running cart complete, we seem to assume that the payment collection is already created.
+    // The payment collection amount should already have the tax amount, why are we updating the tax lines here?
+    updateTaxLinesStep({ cart_or_cart_id: cart, force_tax_calculation: true })
+
+    authorizePaymentSessionStep({
+      id: paymentSession.id,
```

Comment:
Depends on how you treat my other comment:
```suggestion
      id: paymentSessions[0].id,
```